src/basics.pod

```diff
@@ -534,33 +540,31 @@ B<Answer:> Remove the line C<my @names = $file.get.words;>, and change:
 
 =end programlisting
 
-B<2.> Instead of removing the redundancy, you can also use it to warn if a
+B<2.> Instead of deleting the redundant C<@names> variable, you can also use it to warn if a
 player appears that wasn't mentioned in the first line, for example due to a
 typo. How would you modify your program to achieve that?
 
-B<Answer:> Introduce another hash with the names of the legitimate players as
-keys, and look in this hash when the name of a player is read:
+Hint: Try using L<membership operators|https://docs.perl6.org/routine/(elem)>. 
+
+B<Answer:> Change C<@names> to C<@valid-players>. When looping through the lines of
+the file, check to see that C<$p1> and C<$p2> are in C<@valid-players>. Note that
+for membership operators you can also use C<(elem)> and C<!(elem)>.
 
 =begin programlisting
 
     ...
-    my @names = $file.get.split(' ');
-    B<my %legitimate-players;>
-    B<< for @names -> $n { >>
-    B<    %legitimate-players{$n} = 1;>
-    B<}>
-
+    my B<@valid-players> = $file.get.words;
     ...
 
     for $file.lines -> $line {
         my ($pairing, $result) = $line.split(' | ');
         my ($p1, $p2)          = $pairing.split(' ');
-        B<< for $p1, $p2 -> $p { >>
-        B<    if !%legitimate-players{$p} {>
-        B<        say "Warning: '$p' is not on our list!";>
-        B<    }>
-        B<}>
-
+        B< if $p1 ∉ @valid-players {>
```

Comment:
There is a little bit in the Preface that touches on entering Unicode. But in Hint for this exercise I link to the docs on this operator and say you can also use (elem)